internal/handle/go1_21.go

```diff
@@ -0,0 +1,66 @@
+//go:build go1.21
+// +build go1.21
+
+package handle
+
+import "C"
+import (
+	"runtime"
+	"unsafe"
+)
+
+// GoHandle provides a way to pass values that contain Go pointers (pointers to memory allocated by
+// Go) between Go and C without breaking the cgo pointer passing rules. It uses the Pinner feature
+// in the runtime instead of the cgo.Handle type in the cgo runtime to avoid lock contention due to
+// the use of sync.Map in the cgo implementation.
+type GoHandle struct {
+	pinner *runtime.Pinner
+	value  *any
+}
+
+// Value gets the handle pointer.
+func (h *GoHandle) Pointer() uintptr {
+	return uintptr(unsafe.Pointer(h.value))
+}
+
+// Value gets the handle value.
+func (h *GoHandle) Value() any {
+	return *h.value
+}
+
+// Set sets the handle value.
+func (h *GoHandle) Set(v any) {
+	h.Delete()
+	pinner := &runtime.Pinner{}
+	pinner.Pin(pinner)
+	pinner.Pin(&v)
+	h.pinner = pinner
+	h.value = &v
+}
+
+// Delete stops the Go runtime tracking of the referred Go value.
+//
+// If the Go runtime does not manage the GoHandle storage, it cannot detect when the referred value
+// becomes inaccessible. Before deallocating the GoHandle is necessary to close it so to prevent
+// memory leaks.
+func (h *GoHandle) Delete() {
+	if h.pinner != nil {
+		h.pinner.Unpin()
+		h.pinner = nil
+		h.value = nil
+	}
+}
+
+// NewHandle creates a new handle and pins it to the given value.
+func NewHandle(v any) *GoHandle {
```

Comment:
will be good to add some unit tests for GoHandle and Delete like that in the blog post

Reply:
All the existing unit tests broke when the implementation was wrong, so I think there's enough coverage.